File: java/1.java
public class Main {
  public static int strStr(String haystack, String needle) {
    for (int i = 0; i <= haystack.length() - needle.length(); i++) {
      if (haystack.substring(i, i + needle.length()).equals(needle)) {
        return i;
      }
    }
    return -1;
  }

  public static void main(String[] args) {
    String[] haystack = {"bassadbas", "sleetcodes", "github"};
    String[] needle = {"sad", "leetcode", "ggithub"};

    System.out.println(strStr(haystack[0], needle[0]));
    System.out.println(strStr(haystack[1], needle[1]));
    System.out.println(strStr(haystack[2], needle[2]));
  }
}

File: python/1.py
def str_str(haystack, needle):
    for i in range(len(haystack) - len(needle) + 1):
        if haystack[i:i+len(needle)] == needle:
            return i
    return -1

haystack = ["bassadbas", "sleetcodes", "github"]
needle = ["sad", "leetcode", "ggithub"]

print(str_str(haystack[0], needle[0]))
print(str_str(haystack[1], needle[1]))
print(str_str(haystack[2], needle[2]))
